nifi-nar-bundles/nifi-framework-bundle/nifi-stateless-bundle/nifi-stateless-api/src/main/java/org/apache/nifi/stateless/flow/StatelessDataflow.java

```diff
@@ -24,8 +24,33 @@
 import java.util.Set;
 
 public interface StatelessDataflow {
```

Comment:
It is not part of this PR, but wouldn't make sense to make it extend AutoClosable? and having close() instead of shutdown(), etc.

Reply:
I don't think we should extend `AutoCloseable` in this case. The StatelessDataflow is not really a "resource" and is generally not used in a such a way that you'd want to use a try-with-resources to create it for a short period of time and then cleanup. Rather, `shutdown()` is generally used when the application is shutdown. This involves terminating thread pools, cleaning up potentially many resources, etc., etc.